src/validator/YieldValidator.cpp

```diff
@@ -192,6 +201,15 @@ Status YieldValidator::validateWhere(const WhereClause *clause) {
     }
     if (filter != nullptr) {
         NG_RETURN_IF_ERROR(deduceProps(filter, exprProps_));
+        auto newFilter = filter->clone();
+        FoldConstantExprVisitor visitor;
+        newFilter->accept(&visitor);
+        if (visitor.canBeFolded()) {
+            filterCondition_ = visitor.fold(newFilter.get());
+        } else {
+            filterCondition_ = newFilter.release();
+        }
+        qctx_->objPool()->add(filterCondition_);
```

Comment:
IMO. this is tedious and error-prone to use the visitor like this. Could we wrap these into a function, e.g. `OptimizerUtils::foldConstantExpr(Expression*)`

Reply:
I think putting this util function in ExpressionUtils is better than OptimizerUtils. Have fixed it.